main/inc/ajax/course_home.ajax.php

```diff
@@ -131,10 +132,146 @@
+        // Allow tool visibility in sessions.
+        if (api_is_allowed_to_edit(null, true)) {
+            if (is_array($tools_ids) && count($tools_ids) != 0) {
+                $total_tools = count($tools_ids);
+                for ($i = 0; $i < $total_tools; $i++) {
+                    $tool_id = (int)$tools_ids[$i];
+
+                    $criteria = [
+                        'cId' => $course_id,
+                        'sessionId' => 0,
+                        'iid' => $tool_id,
+                    ];
+                    /** @var CTool $tool */
+                    $tool = $repository->findOneBy($criteria);
+                    $visibility = $tool->getVisibility();
+
+                    if ($allowEditionInSession && !empty($sessionId)) {
+                        $criteria = [
+                            'cId' => $course_id,
+                            'sessionId' => $sessionId,
+                            'name' => $tool->getName(),
+                        ];
+
+                        /** @var CTool $tool */
+                        $toolInSession = $repository->findOneBy($criteria);
+                        if ($toolInSession) {
+                            // Use the session
+                            $tool = $toolInSession;
+                            $visibility = $toolInSession->getVisibility();
+                        } else {
+                            // Creates new row in c_tool
+                            $toolInSession = clone $tool;
+                            $toolInSession->setIid(0);
+                            $toolInSession->setId(0);
+                            $toolInSession->setVisibility(0);
+                            $toolInSession->setSessionId($session_id);
+                            $em->persist($toolInSession);
+                            $em->flush();
+                            // Update id with iid
+                            $toolInSession->setId($toolInSession->getIid());
+                            $em->persist($toolInSession);
+                            $em->flush();
+                            // $tool will be updated later
+                            $tool = $toolInSession;
+                        }
+                    }
+
+                    $toolImage = $tool->getImage();
+                    $customIcon = $tool->getCustomIcon();
+
+                    if (api_get_setting('homepage_view') != 'activity_big') {
+                        $toolImage = Display::return_icon(
+                            $toolImage,
+                            null,
+                            null,
+                            null,
+                            null,
+                            true
+                        );
+                        $inactiveImage = str_replace('.gif', '_na.gif', $toolImage);
+                    } else {
+                        // Display::return_icon() also checks in the app/Resources/public/css/themes/{theme}/icons folder
```

Comment:
Codacy found an issue: [Line exceeds maximum limit of 100 characters; contains 113 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)